relay-server/src/actors/store.rs

```diff
@@ -143,23 +143,19 @@ pub struct ShardedProducer {
 
 impl ShardedProducer {
     /// Returns topic name and the Kafka producer based on the provided sharding key.
-    pub fn get_producer(&self, sharding_key: impl Hash) -> (&str, &ThreadedProducer) {
-        let mut hasher = FnvHasher::default();
-        Hash::hash(&sharding_key, &mut hasher);
-        let hashed = hasher.finish();
-        let shard = hashed % self.shards;
-
-        // Safety:
-        // We always have at least 1 shard so it's safe to unwrap this value.
+    /// Returns error [`StoreError::InvalidKafkaShardRange`] if the shard range for the provided sharding
+    /// key could not be found.
+    pub fn get_producer(&self, sharding_key: u64) -> Result<(&str, &ThreadedProducer), StoreError> {
+        let shard = sharding_key % self.shards;
         let (topic_name, producer) = self
             .producers
             .iter()
             .take_while(|(k, _)| *k <= &shard)
             .last()
             .map(|(_, v)| v)
-            .expect("At least one shard is defined");
+            .ok_or(StoreError::InvalidKafkaShardRange)?;
```

Comment:
this is an unrelated change right? we don't think this is now more likely

Reply:
Yes, this change is unrelated. And error should not happen. 
But I also wanted to remove any `unwrap`/`expect` since I already make changes to this function.